### DISCUSSION TITLE
Preventing concurrent queries in v4

### DISCUSSION THREAD

**Discussion by baymac** [TanStack/query#5957] [General] [answered]
Why are there multiple queries with identical keys, and when does this situation arise?

It appears that several components are concurrently utilizing React Query to retrieve data, and they are being re-rendered due to a common state dependency that triggers the query. Is this the likely cause of the issue?

If so, what is the recommended approach for effectively managing React Query when multiple components on a page depend on it?

**Comment by TkDodo** [comment]
> Why are there multiple queries with identical keys, and when does this situation arise?

it cannot, keys have to be unique. if you use the same key, you use the same query. Have a look at the query devtools, they'll tell you what the keys are.

**Reply by baymac** [reply]
I checked devtools and there are infact multiple queries with the same key. Attaching here the screenshot of devtools for ref.

<img width="1037" alt="Screenshot 2023-09-05 at 2 24 30 PM" src="https://github.com/TanStack/query/assets/23079344/2f50c3f8-b9a9-4a94-ab3f-dd5bd0169a97">

You can see `network-id` and `base-address` query has 50 and 3 queries respectively.

**Reply by baymac** [reply]
I am sorry for confusion, just learned these figures are actually observers. But I noticed that the underlying promises are still called multiple times like 3-4 times when it should be just called 1 time.

**Reply by TkDodo** [reply, marked as answer]
with the default staleTime of zero, every mount of a new observer will trigger a background refetch. Customize `staleTime` to reduce the amount of background refetches.

**Reply by baymac** [reply]
this suggestion partially helped. Currently `network-id` query is only called 1 time but `base-address` keeps getting called multiple times like earlier. Is there any other way this could happen?

**Reply by TkDodo** [reply]
would need to see a reproduction to say more

**Reply by baymac** [reply]
@TkDodo I think I figured it out, we had multiple internal calls that contributed to this rather than react query. Thanks for the help. 🙏🏼
